src/osvr/Client/RenderManagerConfig.cpp

```diff
@@ -0,0 +1,190 @@
+    }
+
+    void RenderManagerConfig::parse(const std::string& render_manager_config)
+    {
+      Json::Reader reader;
+      Json::Value root;
+      reader.parse(render_manager_config, root, false);
+
+      m_directMode = root["render_manager_parameters"]["direct_mode"].asBool();
+      m_displayIndex = root["render_manager_parameters"]["direct_display_index"].asUInt();
+      m_directHighPriority = root["render_manager_parameters"]["direct_high_priority"].asBool();
+
+      m_numBuffers = root["render_manager_parameters"]["num_buffers"].asUInt();
+      m_verticalSync = root["render_manager_parameters"]["vertical_sync"].asBool();
+      m_verticalSyncBlockRendering = root["render_manager_parameters"]
+        ["vertical_sync_block_rendering"].asBool();
+      m_windowTitle = root["render_manager_parameters"]["window_title"].asString();
+      m_windowFullScreen = root["render_manager_parameters"]["window_full_screen"].asBool();
+      m_windowXPosition = root["render_manager_parameters"]["window_x_position"].asInt();
+      m_windowYPosition = root["render_manager_parameters"]["window_y_position"].asInt();
+      m_displayRotation = root["render_manager_parameters"]["display_rotation"].asUInt();
+      m_bitsPerColor = root["render_manager_parameters"]["bits_per_color"].asBool();
+
+      m_enableTimeWarp = root["render_manager_parameters"]["time_warp"]["enable"].asBool();
+      m_asynchronousTimeWarp = root["render_manager_parameters"]["time_warp"]["asynchronous_time_warp"].asBool();
+      m_maxMSBeforeVsyncTimeWarp = root["render_manager_parameters"]["time_warp"]["max_ms_before_vsync"].asFloat();
+ 
+      m_renderOverfillFactor = root["render_manager_parameters"]["render_overfill_factor"].asFloat();
+    }
+
+    void RenderManagerConfig::print() const
+    {
+      std::cout << "Direct mode: " << m_directMode << std::endl;
+      std::cout << "Direct mode display index: " << m_displayIndex << std::endl;
+      std::cout << "Number of buffers: " << m_numBuffers << std::endl;
+      std::cout << "Vertical sync: " << m_verticalSync << std::endl;
+      std::cout << "Vertical sync block rendering: " << m_verticalSyncBlockRendering << std::endl;
+      std::cout << "Window Title: " << m_windowTitle << std::endl;
+      std::cout << "Window full screen: " << m_windowFullScreen << std::endl;
+      std::cout << "Window X Position: " << m_windowXPosition << std::endl;
+      std::cout << "Window Y Position: " << m_windowYPosition << std::endl;
+      std::cout << "Display rotation: " << m_displayRotation << std::endl;
+      std::cout << "Bits per color: " << m_bitsPerColor << std::endl;
+      std::cout << "Enable time warp: " << m_enableTimeWarp << std::endl;
+      std::cout << "Asynchronous time warp: " << m_asynchronousTimeWarp << std::endl;
+      std::cout << "Max ms before vsync time warp: " << m_maxMSBeforeVsyncTimeWarp << std::endl;
+      std::cout << "Render overfill factor: " << m_renderOverfillFactor << std::endl;
+    }
```

Comment:
May just be easier to have the parser populate a struct than have a basically-read-only class with lots of accessors.